tests/presets.spec.js

```diff
@@ -1248,6 +1248,809 @@ describe('Presets', function() {
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+        it('should be able to update the version using an anchored literal regex', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            regex: /^version\s*=\s*/,
+            regexFlags: 'm'
+          }, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.versionedFile, 'utf8')).to.equal([
+              'blah blah',
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+        it('should be able to update the version using a simple string pattern', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            regex: 'version = '
+          }, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.versionedFile, 'utf8')).to.equal([
+              'blah blah',
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+        it('should be able to update the version using a string regex', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            regex: 'version\\s*=\\s*'
+          }, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.versionedFile, 'utf8')).to.equal([
+              'blah blah',
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+        it('should be able to update the version using an anchored string regex', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            regex: '^version\\s*=\\s*',
+            regexFlags: 'm'
+          }, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.versionedFile, 'utf8')).to.equal([
+              'blah blah',
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+        it('should be able to update the version using a RegExp object', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            regex: RegExp('version\\s*=\\s*')
```

Comment:
What will happen if we pass `RegExp(/^version\s*=\s*/, 'm')` here - it won't work, right?

I think there could be friction with flags defined in `RegExp` and `regexFlags`. Now that I am thinking maybe we should remove `regexFlags` completely, but provide a default value for flags if a string is passed (e.g. 'm'). Or if `RegExp` is passed, then use its flags. What do you think?

Reply:
The 'm' specified inside the `RegExp` constructor would be lost, as https://stackoverflow.com/a/185529 notes.

> Or if RegExp is passed, then use its flags.

Good idea - I now merge the flags from `regex` and `regexFlags` :smiley:

Reply:
Merging is a nice option.